analytics-api/app/controllers/JobController.scala

```diff
@@ -116,15 +123,68 @@ class JobController @Inject() (
+            // get userId from user auth token
+            val userId = accessTokenValidator.getUserId(userAuthToken.get)
+            var unauthorizedErrMsg = "You are not authorized."
+            APILogger.log("userId retrieved: " + userId)
+            if(!"Unauthorized".equalsIgnoreCase(userId)) {
+                val headers = Map("x-authenticated-user-token" -> userAuthToken.get, "Authorization" -> authBearerToken.getOrElse(""))
+                val userReadResponse = restUtil.get[Response](userApiUrl + userId, Option(headers))
+                APILogger.log("user read response: " + JSONUtils.serialize(userReadResponse))
+                if(userReadResponse.responseCode.equalsIgnoreCase("ok")) {
+                    val userResponse = userReadResponse.result.getOrElse(Map()).getOrElse("response", Map()).asInstanceOf[Map[String, AnyRef]]
+                    val orgDetails = userResponse.getOrElse("rootOrg", Map()).asInstanceOf[Map[String, AnyRef]]
+                    val userRoles = userResponse.getOrElse("organisations", List()).asInstanceOf[List[Map[String, AnyRef]]]
+                      .map(f => f.getOrElse("roles", List()).asInstanceOf[List[String]]).flatMap(f => f)
+                    if (userRoles.filter(f => authorizedRoles.contains(f)).size > 0) {
+                        if (superAdminRulesCheck) {
+                            val userSlug = orgDetails.getOrElse("slug", "").asInstanceOf[String]
+                            APILogger.log("header channel: " + channelId + " org slug: " + userSlug)
+                            if (channelId.equalsIgnoreCase(userSlug)) return (true, None)
+                            else {
+                                // get MHRD tenant value using org search API
+                                val orgSearchApiUrl = config.getString("org.search.url")
+                                val requestBody = """{"request":{"filters":{"channel":"mhrd"},"offset":0,"limit":1000,"fields":["id"]}}"""
+                                val response = restUtil.post[Response](orgSearchApiUrl, requestBody)
+                                APILogger.log("org search response: " + JSONUtils.serialize(response))
+                                val contents = response.result.getOrElse(Map()).getOrElse("response", Map()).asInstanceOf[Map[String, AnyRef]]
+                                  .getOrElse("content", List(Map())).asInstanceOf[List[Map[String, AnyRef]]]
+                                val mhrdChannel = if(contents.size > 0) contents.head.getOrElse("id", "").asInstanceOf[String] else ""
+                                val userChannel = orgDetails.getOrElse("channel", "").asInstanceOf[String]
+                                APILogger.log("user channel: " + userChannel + " mhrd id: " + mhrdChannel)
+                                if (userChannel.equalsIgnoreCase(mhrdChannel)) return (true, None)
```

Comment:
Super admin channel needs to be configured and this needs to be cached on service start

Reply:
Added super admin channel to cache on service start